### ISSUE TITLE
Create new Filters demo page

### ISSUE THREAD

**Issue by mcarrano** [patternfly/patternfly-react#6710]
Create a new demo to collect various filter related demos in one place. List of demos to move to this page TBD but at least want to include two new demos that now live under Text input group.

@mmenestr let's work together to identify what demos should be moved here.

**Comment by mmenestr** [comment]
Preliminary thoughts:
There are a few component examples/demos that should arguably also be shown in the filters demo, including:
- Select list:
> - single select
> - checkbox select
> - multiple select
- Search input 
- Toggle group

Other than that, it would be good to include an attribute filter demo.
This also reminds me that I had created this sketch file with a series of filter demos that we should show: https://www.sketch.com/s/927ca83a-680d-4a69-baaf-c865d58556be

**Comment by mmenestr** [comment]
Some background info on the difference between search input and text input group:

TLDR; Search input and Text input group have the same functionality, but with text input group you can customize the appearance (ie: you can take away the search icon, you can move it to the right etc etc), where as the search input component has the formatting that we'd recommend (from design) but doesn't have any flexibility beyond that.

"ideally, we will use the text input group to build the search input component! so really not much! The search input came first, but as the asks got more complicated we decided to break down the pieces of it and build the text input group so that people could compose their own search inputs and text inputs"

Examples: Right now, if you want to use a SearchInput, the icon is always on the left of the input. But if you use a Text Input Group, you could hide that icon. and the 'submit' button on the Search Input has an arrow in it, but if you use a Text Input Group , you could make that button be any icon you'd like
or, the navigable controls, such as the 'Match with Navigable options' Search input example, are always going to be arranged and styled exactly that way. If you use a Text input group, you can define the navigable controls to use different buttons or arrange them in a different order, or anything you'd like

now, I don't know if all these types of adjustments would be officially 'endorsed' by design, but basically the search input is like an endorsed packaged solution for when you want to use the Text Input Group as a search input

**Comment by mcarrano** [comment]
Let's copy/link these examples rather than move them if they already exist elsewhere. For others these may be new demos.

**Comment by nicolethoen** [comment]
Other possible filterable examples/demos:
- with a simple attribute value search
- with an advanced search
- look at the existing filterable table demo (under Table -> React demos)
- Look at existing Toolbar examples

**Comment by mmenestr** [comment]
Marvel mocks: https://marvelapp.com/prototype/7i0dj2d

**Comment by tlabaj** [comment]
closed by #7949
